geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala

```diff
@@ -87,11 +82,16 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     val temporal = extractTemporal(dtgField)(temporalFilters)
     val interval = netInterval(temporal)
     val geometryToCover = netGeom(collectionToCover)
-    val filter = buildFilter(geometryToCover, interval)
 
-    output(s"GeomsToCover: $geomsToCover")
+    //This is correctly doing an AcceptEverythingFilter if the geometryToCover is the whole world
+    val filter = buildFilter(geometryToCover, interval)
+    // This catches the case when a whole world query slips through DNF/CNF
+    // The union on this geometry collection is necessary at the moment but is not true
+    // If given spatial predicates like disjoint.
+    val ofilter = if (geometryToCover != null && isWholeWorld(geometryToCover.union)) {
```

Comment:
I'd suggest adding a null check to isWholeWorld so that this call can be simplified.